docsite/rst/playbooks_tests.rst

```diff
@@ -0,0 +1,165 @@
+
+'match' will require a complete match in the string, while 'search' will require a match inside of the string.
+
+
+.. _testing_versions:
+
+Version Comparison
+------------------
+
+.. versionadded:: 1.6
+
+To compare a version number, such as checking if the ``ansible_distribution_version``
+version is greater than or equal to '12.04', you can use the ``version_compare`` filter.
+
+The ``version_compare`` filter can also be used to evaluate the ``ansible_distribution_version``::
+
+    {{ ansible_distribution_version | version_compare('12.04', '>=') }}
+
+If ``ansible_distribution_version`` is greater than or equal to 12, this filter will return True, otherwise it will return False.
+
+The ``version_compare`` filter accepts the following operators::
+
+    <, lt, <=, le, >, gt, >=, ge, ==, =, eq, !=, <>, ne
+
+This test also accepts a 3rd parameter, ``strict`` which defines if strict version parsing should
+be used.  The default is ``False``, and if set as ``True`` will use more strict version parsing::
+
+    {{ sample_version_var | version_compare('1.0', operator='lt', strict=True) }}
+
+
+.. _math_tests:
+
+Group theory tests
+------------------
+
+To see if a list includes or is included by another list you can use 'issubset' and 'issuperset'::
+
+    vars:
+        a: [1,2,3,4,5]
+        b: [2,3]
+    tasks:
+        - debug: msg="A includes B"
+          when: a|issuperset(b)
+
+        - debug: msg="B is included in A"
+          when: b|issubset(a)
+
+
+.. _path_tests:
+
+Testing paths
+-------------
+
+These are tests that can give you information about a path on the controller::
+
+    - debug: msg="path is a directory"
+      when: mypath|isdir
+
+    - debug: msg="path is a file"
+      when: mypath|is_file
+
+    - debug: msg="path is a symlink"
+      when: mypath|is_link
+
+    - debug: msg="path already exists"
+      when: mypath|exists
+
+    - debug: msg="path is {{ (mypath|is_abs)|ternary('absolute','relative')}}"
+
+    - debug: msg="path is the same file as path2"
+      when: mypath|samefile(path2)
+
+    - debug: msg="path is a mount"
+      when: mypath|ismount
+
+
+.. _test_task_results:
+
+Task results
+------------
+
+The following tasks are illustrative of the tests meant to check the status of tasks::
+
+    tasks:
+
+      - shell: /usr/bin/foo
+        register: result
+        ignore_errors: True
+
+      - debug: msg="it failed"
+        when: result|failed
+
+      # in most cases you'll want a handler, but if you want to do something right now, this is nice
+      - debug: msg="it changed"
+        when: result|changed
+
+      - debug: msg="it succeeded in Ansible >= 2.1"
+        when: result|succeeded
+
+      - debug: msg="it succeeded"
+        when: result|success
+
+      - debug: msg="it was skipped"
+        when: result|skipped
+
+.. note:: From 2.1 You can also use success, failure, change, skip so the grammer matches, for those that want to be strict about it.
```

Comment:
From 2.1, you can also use success, failure, change, and skip so that the grammar matches, for those who need to be strict about it.